### PR TITLE
Java REST tests: more version -> feature update

### PR DESCRIPTION
**Pull request by ldematte**
Slowly chipping away `isOriginalClusterVersionAtLeast` usages

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java [hunk 1]
```diff
@@ -2107,7 +2110,53 @@ protected static IndexVersion minimumIndexVersion() throws IOException {
...
+        assertNotNull(minTransportVersion);
+        return minTransportVersion;
+    }
+
+    protected static TransportVersion getTransportVersionWithFallback(
+        Object versionField,
+        Object transportVersionField,
+        Supplier<TransportVersion> fallbackSupplier
+    ) {
+        if (transportVersionField instanceof Integer transportVersionId) {
```
Comment (thecoop) [inline test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java:2143]: This should be checking against `Number`, json can be funny with number types sometimes

File: x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/ApiKeyBackwardsCompatibilityIT.java [hunk 2]
```diff
@@ -304,10 +304,16 @@ private static String randomRoleDescriptors(boolean includeRemoteIndices) {
     }
 
     boolean nodeSupportApiKeyRemoteIndices(Map<String, Object> nodeDetails) {
-        // TODO[lor]: the method can be kept, but we need to replace version check with features checks
-        String versionString = (String) nodeDetails.get("version");
-        Version version = Version.fromString(versionString.replace("-SNAPSHOT", ""));
-        return version.onOrAfter(API_KEY_SUPPORT_REMOTE_INDICES_VERSION);
+        var nodeVersion = nodeDetails.get("version");
```
Comment (thecoop) [inline x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/ApiKeyBackwardsCompatibilityIT.java:307]: Could we use more explicit types here, to make it clearer what types we're working with here?

File: x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/SearchableSnapshotsRollingUpgradeIT.java [hunk 3]
```diff
@@ -288,7 +288,7 @@ private void executeBlobCacheCreationTestCase(Storage storage, long numberOfDocs
             assertHitCount(index, equalTo(numberOfDocs * 2L));
             deleteIndex(index);
 
-            if (isOriginalClusterVersionAtLeast(Version.V_7_13_0)) {
```
[the remark this answers is not in the document]
Comment (ldematte) [reply x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/SearchableSnapshotsRollingUpgradeIT.java:291]: You are right, let me double check. Also true for `RestTestLegacyFeatures.SECURITY_GRANT_API_SUPPORTED`; may not be needed and I can just remove the check.

File: test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java [hunk 4]
```diff
@@ -2136,18 +2136,18 @@ protected static TransportVersion minimumTransportVersion() throws IOException {
...
     protected static TransportVersion getTransportVersionWithFallback(
-        Object versionField,
+        String versionField,
         Object transportVersionField,
         Supplier<TransportVersion> fallbackSupplier
     ) {
-        if (transportVersionField instanceof Integer transportVersionId) {
-            return TransportVersion.fromId(transportVersionId);
+        if (transportVersionField instanceof Number transportVersionId) {
+            return TransportVersion.fromId((int) transportVersionId);
```
Comment (thecoop) [inline test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java:2144]: This is still assuming its going to be an `Integer`. You should use `Number.intValue` instead

**Comment by ldematte** [comment]
@elasticsearchmachine rerun elasticsearch-ci/bwc-snapshots

**Review by thecoop — APPROVED** [review]

**Pull request merged**